src/CarbonAware.WebApi/test/integrationTests/CarbonAware.WebApi.IntegrationTests.csproj

```diff
@@ -7,12 +7,13 @@
   </PropertyGroup>
 
   <ItemGroup>
-    <PackageReference Include="Microsoft.NET.Test.Sdk"/>
-    <PackageReference Include="Moq"/>
-    <PackageReference Include="NUnit"/>
-    <PackageReference Include="NUnit3TestAdapter"/>
+    <PackageReference Include="Microsoft.NET.Test.Sdk" />
+    <PackageReference Include="Moq" />
+    <PackageReference Include="NUnit" />
+    <PackageReference Include="NUnit3TestAdapter" />
     <PackageReference Include="Microsoft.AspNetCore.Mvc.Testing" Version="6.0.0" />
-    <PackageReference Include="WireMock.Net"/>
+    <PackageReference Include="Snapper.Nunit" Version="2.4.0" />
```

Comment:
This is a new package. Just by looking at how it is used, I'm wondering if there is any benefit of using this compared of having a static file with the results. Still very demanding, hard to maintain imho, since the output of the other files were generated with this new tool apparently.

Reply:
It's called snapshot testing. The way it works: when you first write your testcase and use AssetThat Matches.ChildSnapshot() the test runner will check if the snapshot file is present. If the snapshot file is not present it's generated automatically, so you don't need to create it manually. If then, for example, the underlying implementation changes with braking changes, so the output now doesn't match the snapshot, you can update/regenerate the snapshot by adding [[UpdateSnapshots]](https://theramis.github.io/Snapper/#/pages/snapper/updating_snapshots) attribute to a test and running it once, so the generation of the snapshot test is automatic. So it is much less demanding to write/update tests in this way.

Having a static file on the other hand means that we need:
1. To write additional code that parses the file (possibly matching the same format as the snapper library) and matches the test cases to the expected result
2. Write the test cases output files ourselves manually
3. Update/regenerate test cases output files manually if something changes
With snapshot testing we don't need to do this steps